pkg/steps/index_generator.go

```diff
@@ -89,9 +93,10 @@ func (s *indexGeneratorStep) run(ctx context.Context) error {
 func (s *indexGeneratorStep) indexGenDockerfile() (string, error) {
 	var dockerCommands []string
 	dockerCommands = append(dockerCommands, "FROM quay.io/operator-framework/upstream-opm-builder AS builder")
-	// pull secret is needed for opm command
-	dockerCommands = append(dockerCommands, "COPY .dockerconfigjson .")
-	dockerCommands = append(dockerCommands, "RUN mkdir $HOME/.docker && mv .dockerconfigjson $HOME/.docker/config.json")
+	if s.pullSecret != nil {
+		dockerCommands = append(dockerCommands, "COPY .dockerconfigjson .")
+		dockerCommands = append(dockerCommands, "RUN mkdir $HOME/.docker && mv .dockerconfigjson $HOME/.docker/config.json")
+	}
```

Comment:
```suggestion
	} else {
	    logrus.Warn("The pull secret is nil in indexGeneratorStep which is probably a result of a non-generated job and thus not mounted onto the build pod")
	}
```

As you pointed out, this should not happen in the generated jobs.
But I would like to log it if it  does happen, to reduce the debugging time.

Reply:
Answering this and the top-level question ("do we want to understand it") here: the comment is incorrect — the secret is not necessary for the `opm` command.

It may be necessary for the specific setup we have (or some of them?), but it's entirely possible to use this functionality without needing a pull secret.

Reply:
I think the logging is useful but that is NIT.

Still not clear when the pull secret is needed.
I understand it is not in the scope of this PR.